independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTreeVisit.java

```diff
@@ -17,7 +17,10 @@ static void walk(Path root, Path rootDir, Path walkDir, PathFilter pathFilter, M
             PathVisitor visitor) {
         final PathTreeVisit visit = new PathTreeVisit(root, rootDir, pathFilter, multiReleaseMapping);
         try (Stream<Path> files = Files.walk(walkDir)) {
-            final Iterator<Path> i = files.iterator();
+            final Iterator<Path> i = files
+                    // we sort the elements to be deterministic, we compare the toString() to make sure the order is the same on Linux and Windows
+                    .sorted((p1, p2) -> p1.toString().compareTo(p2.toString()))
```

Comment:
Does this appear to be generally necessary or only in specific cases?

Reply:
So it's probably only necessary in specific cases... but figuring out which cases is hard and when you had new use cases, people will have to think about it (and will forget).

I think in general bringing a more deterministic behavior is nice.

As for the specific comparator, it's the only way to get the same ordering on Linux and Windows.

Now, maybe we shouldn't merge this PR for now and wait for the whole thing and then try to evaluate the cost of it?